src/controllers/listingsOrganisations.ts

```diff
@@ -0,0 +1,99 @@
+import { db } from  '../database/db';
+import { asyncHandler } from '../middleware';
+import { cleanseData, ErrorResponse, parseSqlUpdateStmt } from '../utils';
+
+/**
+ * @desc    Create an entry in the listings-organisations table
+ * @route   POST /api/listings-organisations
+ * @access  Private
+ */
+export const createListingOrganisation = asyncHandler(async (req, res) => {
+    const { listing_id, organisation_id } = req.body;
+    const data = {
+        listing_id,
+        organisation_id
+    };
+
+    cleanseData(data);
+
+    const rows = await db.one('INSERT INTO listingsorganisations VALUES (${this:csv}) RETURNING *', data);
+
+    res.status(201).json({
+        success: true,
+        data: rows,
+    });
+});
+
+/**
+ * @desc    Update an entry in the listings-organisations table
+ * @route   PUT /api/listings-organisations/:id
+ * @access  Admin/Owner
+ */
+export const updateListingOrganisation = asyncHandler(async (req, res, next) => {
+    const userId: string = req.user.user_id;
+    const ids = await db.one('SELECT listing_id, organisation_id FROM listingsorganisations WHERE listing_organisation_id = $1', req.params.id);
+    const table_listing_id: string = ids[0];
+    const table_organisation_id: number = ids[1];
+
+    const isOrganisationOwner = checkOrganisationOwner(userId, table_organisation_id);
+    const isListingOwner = checkListingOwner(userId, table_listing_id);
+    if (req.user.role !== 'admin' && !isListingOwner && !isOrganisationOwner) {
+        return next(new ErrorResponse('Not authorised to update listing organisation as you are not the organisation or listing owner', 403));
+    }
+
+    const {
+        listing_id,
+        organisation_id
+    } = req.body;
+
+    const data = {
+        listing_id,
+        organisation_id
+    };
+
+    cleanseData(data);
+
+    const updateListingOrganisationQuery = parseSqlUpdateStmt(data, 'organisations', 'WHERE organisation_id = $1 RETURNING *', [req.params.id]);
+
+    const rows = db.one(updateListingOrganisationQuery);
+
+    res.status(200).json({
+        success: true,
+        data: rows,
+    });
+});
```

Comment:
Actually, i think that the update endpoint wont be needed, as from the frontend perspective, when a listing joins a wrong organisation, it does not update the `organisation_id` it joined. 

Rather, it will leave the organisation, and join the correct organisation. This will only require the create and delete endpoint.